code/modules/projectiles/guns/ballistic/automatic.dm

```diff
@@ -352,3 +352,192 @@
+	fire_delay = initial(fire_delay)
+	update_appearance()
+	balloon_alert(user, "system reset.")
+
+/obj/item/gun/ballistic/automatic/battle_rifle/try_fire_gun(atom/target, mob/living/user, params)
+	. = ..()
+	if(!chambered || chambered && !chambered.loaded_projectile)
+		return
+
+	if(shots_before_degradation)
+		shots_before_degradation --
+		return
+
+	else if (obj_flags & EMAGGED && degradation_stage == degradation_stage_max && !explosion_timer)
+		perform_extreme_malfunction(user)
+
+	else
+		attempt_degradation(FALSE)
+
+
+/obj/item/gun/ballistic/automatic/battle_rifle/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
+	if(chambered.loaded_projectile && emp_malfunction && prob(75) || chambered.loaded_projectile && degradation_stage == degradation_stage_max && prob(75))
+		balloon_alert_to_viewers("*click*")
+		playsound(src, dry_fire_sound, dry_fire_sound_volume, TRUE)
+		return
+
+	return ..()
+
+/// Proc to handle weapon degradation. Called when attempting to fire or immediately after an EMP takes place.
+/obj/item/gun/ballistic/automatic/battle_rifle/proc/attempt_degradation(force_increment = FALSE)
+	if(!prob(degradation_probability) && !force_increment || degradation_stage == degradation_stage_max)
+		return //Only update if we actually increment our degradation stage
+
+	degradation_stage = clamp(degradation_stage + (obj_flags & EMAGGED ? 2 : 1), 0, degradation_stage_max)
+	projectile_speed_multiplier = clamp(initial(projectile_speed_multiplier) + degradation_stage * 0.1, initial(projectile_speed_multiplier), maximum_speed_malus)
+	fire_delay = initial(fire_delay) + (degradation_stage * 0.5)
+	do_sparks(1, TRUE, src)
+	update_appearance()
+
+/// Called by /obj/machinery/recharger while inserted: attempts to recalibrate our gun but reducing degradation.
+/obj/item/gun/ballistic/automatic/battle_rifle/proc/attempt_recalibration(restoring_shots_before_degradation = FALSE, recharge_rate = 1)
+	if(emp_malfunction)
+		emp_malfunction = FALSE
+
+	if(restoring_shots_before_degradation)
+		shots_before_degradation = clamp(round(shots_before_degradation + recharge_rate, 1), 0, max_shots_before_degradation)
+
+	else
+		degradation_stage = clamp(degradation_stage - 1, 0, degradation_stage_max)
+		if(degradation_stage)
+			projectile_speed_multiplier = clamp(initial(projectile_speed_multiplier) - degradation_stage * 0.1, maximum_speed_malus, initial(projectile_speed_multiplier))
+			fire_delay = initial(fire_delay) + (degradation_stage * 0.5)
+		else
+			projectile_speed_multiplier = initial(projectile_speed_multiplier)
+			fire_delay = initial(fire_delay)
+
+	update_appearance()
+
+/// Proc to handle the countdown for our detonation
+/obj/item/gun/ballistic/automatic/battle_rifle/proc/perform_extreme_malfunction(mob/living/user)
+	balloon_alert(user, UNLINT("HOLY FUCK THE GUN IS GOING TO EXPLODE, THROW IT, THROW IT NOW!"))
```

Comment:
Little goofy

If you had a problem with people not noticing it was going to explode, I would maybe recommend looping through alerts before it blows

```dm
balloon_alert("gun is exploding!") 
// (1 second timer)
if(!thrown)
	balloon_alert("gun is exploding!") 
// (1 second timer)
if(!thrown)
	balloon_alert("gun is exploding, throw it!") 
// (1 second timer)
if(!thrown)
	balloon_alert("gun is exploding, throw it!") 
// (1 second timer)
if(!thrown)
	balloon_alert("GUN IS ABOUT TO EXPLODE!") 
// 1 second timer
explode()
```

Reply:
It was mostly just as a joke more than any practical purpose.

Reply:
I've at the very least shortened the message.